src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHandler.cs

```diff
@@ -53,17 +49,26 @@ private bool TrySetBasicAuthToken(HttpRequestMessage request)
 
                 foreach (AuthenticationHeaderValue h in authenticateValues)
                 {
-                    // We only support Basic auth, ignore others
-                    if (h.Scheme == Basic)
+                    // We only support Basic and digest auth, ignore others
```

Comment:
@mconnew How do i recognise that scenario here? Should i check if response is Unauthorized and try request again with new nonce? How many times to repeat this?